apis/intermediate_api/intermediate_api/utils.py

```diff
@@ -67,52 +102,106 @@ def proxy_to_api(
             )
 
-    if methods == "POST":
-        resp = requests.post(
-            api_destination + route, data=data, headers={"Authorization": api_token},
+        return flask.Response(resp.content, resp.status_code, mimetype=content_type)
+    else:
+        return (
+            flask.jsonify({"Warning": "You do not have permission to access."}),
+            const.UNAUTHORIZED_CODE,
         )
 
-    elif methods == "GET":
-        querystring = request.query_string.decode("utf-8")
 
-        if querystring:
-            querystring = "?" + querystring
+def proxy_to_project_api(
+    request,
+    route,
+    api_destination: str,
+    api_token: str,
+    authorized: bool = False,
+    user_id: str = None,
+    action: str = None,
+    content_type: str = "application/json",
+):
+    """IntermediateAPI - Handling the communication between Frontend and Core/Project API.
 
+    Parameters
+    ----------
+    request: object
+    route: string
+        URL path to Core/Project API
+    api_destination: string
+        To determine the destination, either the CoreAPI or ProjectAPI
+    api_token: string
+        Special token to pass the CoreAPI/ProjectAPI's authorization check
+    authorized: boolean
+        True if it the request is with Auth0 token which is for private
+        otherwise, public access
+    user_id: string
+        Determining the user
+    action: string
+        To find out what user is performing
+    content_type: string
+        Entry-header field indicates the media type of the entity-body sent to the recipient.
+        The default media type is application/json
+    """
+    if action and user_id:
+        records_user_activity(request, action, api_destination, user_id)
+
+    querystring = request.query_string.decode("utf-8")
+
+    if querystring:
+        querystring = "?" + querystring
+
+    # Getting a list of project ids - authentication check is done within
+    # get_available_project_ids with project table
+    if querystring == "" and not authorized:
         resp = requests.get(
             api_destination + route + querystring, headers={"Authorization": api_token},
         )
-
-    return flask.Response(resp.content, resp.status_code, mimetype=content_type)
+        return flask.Response(resp.content, resp.status_code, mimetype=content_type)
+
+    # Every call that is not getting a list of project ids, have an argument with project_id
+    project_id = request.args.get("project_id")
+    # One exception with download as it only has token which contains the project id information
+    if "/download" in route:
+        project_id = get_token_payload(
+            list(request.args.to_dict().values())[0], DOWNLOAD_URL_SECRET_KEY_PROJECT
+        ).get("project_id")
+
+    # Authorized and requested project id is in the Project table
+    # Not authorized but requested project id is in the Project table with Public access_level
+    if (authorized and project_id in db.get_projects()) or (
```

Comment:
I think I get what you are doing here (in the whole function, not this line specifically), but I don't think its a good.
This function proxies, it shouldn't do permission checking, it might save some lines, but its unclear and tricky to read imo, making it hard to update, maintain and error prone. I suggest the permission checking is done in each endpoint.

Also specifically for this line, it checks that the user is authenticated and that the project exists, but it doesn't check if the user actually is allowed access to the project.